src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -157,6 +169,24 @@ public function __construct(ClassMetadataFactoryInterface $classMetadataFactory
         if (isset($this->defaultContext[self::CIRCULAR_REFERENCE_HANDLER]) && !\is_callable($this->defaultContext[self::CIRCULAR_REFERENCE_HANDLER])) {
             throw new InvalidArgumentException(sprintf('Invalid callback found in the "%s" default context option.', self::CIRCULAR_REFERENCE_HANDLER));
         }
+
+        if (null === $instantiator) {
+            $instantiator = new Instantiator($classMetadataFactory, $classDiscriminatorResolver, $propertyTypeExtractor, null, $nameConverter, null);
+
+            if ($this->denormalizer instanceof DenormalizerInterface) {
+                $instantiator->setDenormalizer($this->denormalizer);
+            }
+        }
+        $this->instantiator = $instantiator;
+    }
+
+    public function setDenormalizer(DenormalizerInterface $denormalizer)
+    {
+        $this->denormalizer = $denormalizer;
+
+        // Because we need a denormalizer in the Instantiator and we create it in the construct method, it won't get it.
+        // So we are obliged to overwrite this method in order to give the denormalizer to the Instantiator.
+        $this->instantiator->setDenormalizer($denormalizer);
     }
```

Comment:
This is the strange part. `Instantiator` needs a denormalizer to work, this is why this class has the `DenormalizerAwareInterface` interface. But since the `Instantiator` can be created in the `__construct` so it will never receive a denormalizer (the `DenormalizerAwareInterface` will be catch either by the `Serializer` or by the DIC, here none of them will be triggered).
So in order to still get that denormalizer, we overwrite the `setDenormalizer` method here and give the denormalizer to the `Instantiator`.